Sample Applications/WPFGallery/MainWindow.xaml.cs

```diff
@@ -250,4 +239,31 @@ private void ControlsList_Loaded(object sender, RoutedEventArgs e)
             }
         }
     }
+
+    private void SearchBox_PreviewKeyDown(object sender, KeyEventArgs e)
+    {
+        var comboBox = sender as ComboBox;
+
+        if(!string.IsNullOrWhiteSpace(comboBox.Text))
+        {
+            comboBox.IsDropDownOpen = true;
+        }
+    }
+
+    private void SearchBox_DropDownClosed(object sender, EventArgs e)
+    {
+        var comboBox = sender as ComboBox;
+
+        if (comboBox.SelectedItem != null)
+        {
+            _navigationService.Navigate(comboBox.SelectedItem.ToString());
+        }
+    }
+
+    private void SearchBox_GotFocus(object sender, RoutedEventArgs e)
+    {
+        var comboBox = sender as ComboBox;
+
+        comboBox.IsDropDownOpen = true;
```

Comment:
What is the behavior when we navigate on the drop down using arrow keys ? For A11y, we will have to avoid selection, but I am guessing that's working correctly. I will try to run the changes and check it out. Overall, good work.